explorer/interpreter/interpreter.cpp

```diff
@@ -423,30 +424,32 @@ auto Interpreter::StepLvalue() -> ErrorOr<Success> {
         //    { {e.f :: C, E, F} :: S, H}
         // -> { e :: [].f :: C, E, F} :: S, H}
         return todo_.Spawn(std::make_unique<LValAction>(&access.object()));
-      } else {
-        if (auto constant_value = access.constant_value()) {
-          CARBON_ASSIGN_OR_RETURN(
-              Nonnull<const Value*> instantiated,
-              InstantiateType(*constant_value, access.source_loc()));
-          return todo_.FinishAction(instantiated);
+      } else if (act.pos() == 1) {
+        auto constant_value = access.constant_value();
+        if (constant_value) {
```

Comment:
nit: Prefer the "if with initializer" form to avoid bringing that variable into scope